cmd/kpod/launch.go

```diff
@@ -1,16 +1,553 @@
 package main
 
 import (
+	"encoding/hex"
 	"fmt"
+	"os"
+	"strings"
 
+	"github.com/Sirupsen/logrus"
+	"github.com/kubernetes-incubator/cri-o/server"
+	"github.com/opencontainers/runc/libcontainer/selinux"
 	"github.com/urfave/cli"
+	"golang.org/x/net/context"
+	pb "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
 )
 
-// TODO implement
+// LAUNCH TODO
+// Remove container (and sandbox if we created it) - high priority for usability & testing
+// Terminal attach implementation (kpod attach command?)
+// Logging (interaction with crio daemon?)
+// Properly place created containers in cgroups
+// Sanely populate metadata for sandbox
+// Missing parsing in CLI handling - DNS, port forwards, mounts, devices, resource limits etc
+// Labels and Annotations (pod & container)
+// Security & confinement - SELinux, AppArmor, seccomp, capabilities, run as users
+// Interface with crio daemon - locking to prevent trampling pod status
+// Launch containers in existing sandboxes
+// Integration tests
+// Man pages
+
 var launchCommand = cli.Command{
 	Name:  "launch",
 	Usage: "launch a pod",
-	Action: func(context *cli.Context) error {
-		return fmt.Errorf("this functionality is not yet implemented")
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "image",
+			Value: "",
+			Usage: "image to launch",
+		},
+		cli.BoolFlag{
+			Name:  "attach",
+			Usage: "attach to the container once it is created",
```

Comment:
"Primary container" sounds like a good way of phrasing it, I think. We can specify that all operations that target a single container will affect only the primary.